lib/src/logic/actions/move_tile_action.dart

```diff
@@ -5,20 +5,23 @@ import 'package:link_five/src/model/game/tile.dart';
 import 'package:link_five/src/model/game/tile_location.dart';
 
 class MoveTileAction extends GameAction {
-  TileLocation source;
-  TileLocation destination;
+  final TileLocation _source;
+  final TileLocation _destination;
+  TileLocation get source => _source;
+  TileLocation get destination => _destination;
```

Comment:
Since these are final there's no need to have separate private variables. There's no corresponding setter by default because they are final.
```suggestion
  final TileLocation source;
  final TileLocation destination;
```